### PR TITLE
Blender: Implement USD Extractor and Loader

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Implement simple USD extract and load for Blender

## Additional info

This does not contain logic yet to 'contribute' the USD data into another USD Asset layer. That will be a separate PR.

## Testing notes:

1. Extract USD geometry from Blender
2. Load USD to Blender
3. Load Alembic to Blender should still work
4. Updating loaded content should work

### CODE REVIEW

File: client/ayon_core/hosts/blender/plugins/load/load_abc.py
Comment (moonyuet) [inline client/ayon_core/hosts/blender/plugins/load/load_abc.py:1]: should we rename this as `load_cache_model.py` as it no longer only supports alembic loader.

File: client/ayon_core/hosts/blender/plugins/publish/extract_usd.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                export_textures=False,
+                relative_paths=False,
+                export_animation=False,
+                export_hair=False,
+                export_uvmaps=True,
+                # TODO: add for new version of Blender (4+?)
+                # export_mesh_colors=True,
+                export_normals=True,
+                export_materials=True,
+                use_instancing=True
```
[the remark this answers is not in the document]
Comment (BigRoy) [reply client/ayon_core/hosts/blender/plugins/publish/extract_usd.py:65]: I'd leave that for a separate PR - if you're ok with that.

[the remark this answers is not in the document]
Comment (moonyuet) [reply client/ayon_core/hosts/blender/plugins/publish/extract_usd.py:65]: Sure things!

**Review by moonyuet — APPROVED** [review]
Tested successfully in Blender. I also tried the loader with both model and usd family to make sure there is no bug encountered during the load too.

**Comment by LiborBatek** [comment]
@BigRoy just started to "investigate" this PR and as I pulled it I dont see any Blender Addon Extractors settings for USD. is that correct? thx

image

I have built the addon package for blender addon (ver 0.1.8) or?

**Comment by BigRoy** [comment]
> @BigRoy just started to "investigate" this PR and as I pulled it I dont see any Blender Addon Extractors settings for USD. is that correct? thx
> 
> image
> 
> I have built the addon package for blender addon (ver 0.1.8) or?

Correct - this does not expose any settings for these extractors? What settings would you expect? Just whatever `Extract FBX` also exposes?

**Review by LiborBatek — CHANGES_REQUESTED** [review]
The USD creator / extractor is available within ```Publish``` but even if not select anything it does ```Validate``` ok even being empty.

Another issue is when using ```Load``` that as ```Alembic``` and ```USD``` being on the same loader user cannot choose which ```representation``` should be loaded which is not ideal...as now user cant go with ```Alembic``` ?
Screenshot 2024-04-19 141210

Or am I missing something?

**Comment by LiborBatek** [comment]
> > @BigRoy just started to "investigate" this PR and as I pulled it I dont see any Blender Addon Extractors settings for USD. is that correct? thx
> > image
> > I have built the addon package for blender addon (ver 0.1.8) or?
> 
> Correct - this does not expose any settings for these extractors? What settings would you expect? Just whatever `Extract FBX` also exposes?

Well to be able to ```Disable``` it for example :)

**Comment by BigRoy** [comment]
@LiborBatek tweaked some bits - let me know if you still get confused. 
The `model` family can now also publish USD and will also do so by default - configurable in settings.

**Comment by BigRoy** [comment]
Ready to merge?

**Pull request merged**